create_file(..., contents=None) unintentionally crates a "large file"

Seems it's a lack in the API:
```python
    # pytest
    def test_thinks_is_a_large_file(fs):
        fs.create_file("/tmp/file.txt", contents=None)
    
        with open("/tmp/file.txt", 'rt') as ff:
            ff.read()
```
If the contents is `None` there is no chance to later read the file.
In my opinion `None` in this API should be equivalent to `""` - giving an empty file. However at read attempt it raises:

```python
    self = <pyfakefs.fake_filesystem.FakeFileWrapper object at 0x7f5a8c6dd310>
    name = 'read'
    
        def __getattr__(self, name):
            if self.file_object.is_large_file():
    >           raise FakeLargeFileIoException(self._file_path)
    E           FakeLargeFileIoException: Read and write operations not supported for fake large file: /tmp/file.txt
```
Of course setting `contents=''` works OK.

--- a/pyfakefs/fake_filesystem.py
+++ b/pyfakefs/fake_filesystem.py
@@ -2259,8 +2259,11 @@ class FakeFilesystem(object):
         Args:
             file_path: The path to the file to create.
             st_mode: The stat constant representing the file type.
-            contents: The contents of the file.
-            st_size: The file size; only valid if contents not given.
+            contents: the contents of the file. If not given and st_size is
+                None, an empty file is assumed.
+            st_size: file size; only valid if contents not given. If given,
+                the file is considered to be in "large file mode" and trying
+                to read from or write to the file will result in an exception.
             create_missing_dirs: If `True`, auto create missing directories.
             apply_umask: `True` if the current umask must be applied
                 on `st_mode`.
@@ -2413,8 +2416,11 @@ class FakeFilesystem(object):
         Args:
             file_path: path to the file to create.
             st_mode: the stat.S_IF constant representing the file type.
-            contents: the contents of the file.
-            st_size: file size; only valid if contents not given.
+            contents: the contents of the file. If not given and st_size is
+                None, an empty file is assumed.
+            st_size: file size; only valid if contents not given. If given,
+                the file is considered to be in "large file mode" and trying
+                to read from or write to the file will result in an exception.
             create_missing_dirs: if True, auto create missing directories.
             apply_umask: whether or not the current umask must be applied
                 on st_mode.
@@ -2456,6 +2462,8 @@ class FakeFilesystem(object):
         file_object.st_ino = self._last_ino
         self.add_object(parent_directory, file_object, error_fct)
 
+        if st_size is None and contents is None:
+            contents = ''
         if (not read_from_real_fs and
                 (contents is not None or st_size is not None)):
             try:

--- a/pyfakefs/tests/fake_filesystem_test.py
+++ b/pyfakefs/tests/fake_filesystem_test.py
@@ -543,6 +543,13 @@ class FakeFilesystemUnitTest(TestCase):
         self.assertTrue(stat.S_IFREG & new_file.st_mode)
         self.assertEqual(new_file, retval)
 
+    def test_empty_file_created_for_none_contents(self):
+        fake_open = fake_filesystem.FakeFileOpen(self.filesystem)
+        path = 'foo/bar/baz'
+        self.filesystem.create_file(path, contents=None)
+        with fake_open(path) as f:
+            self.assertEqual('', f.read())
+
     def test_create_file_with_incorrect_mode_type(self):
         self.assertRaises(TypeError, self.filesystem.create_file, 'foo', 'bar')
 

--- a/pyfakefs/tests/fake_os_test.py
+++ b/pyfakefs/tests/fake_os_test.py
@@ -4693,7 +4693,7 @@ class FakeScandirTest(FakeOsModuleTestBase):
         link_path = self.make_path('A', 'C')
         self.os.symlink(dir_path, link_path)
         self.assertEqual([self.os.path.join(link_path, 'D')],
-                         [f.path for f in self.os.scandir(link_path)])
+                         [f.path for f in self.scandir(link_path)])
 
     def test_inode(self):
         if has_scandir and self.is_windows and self.use_real_fs():

